packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+
+import { useCoreClerk, useCoreSession } from '../contexts';
+import { Col, Flex, Icon, Link, Text, useAppearance } from '../customizables';
+import { Portal } from '../elements/Portal';
+import { Eye } from '../icons';
+
+type EyeCircleProps = {
+  width: string;
+  height: string;
+  padding: string;
+};
+
+const EyeCircle: React.FC<EyeCircleProps> = ({ width, height, padding }) => {
+  return (
+    <Flex
+      id='impersonationEye'
+      sx={t => ({
+        padding,
+        transition: 'transform 800ms',
+        backgroundColor: t.colors.$danger500,
+        borderRadius: t.radii.$circle,
+      })}
+    >
+      <Icon
+        icon={Eye}
+        sx={{
+          width,
+          height,
+          transform: 'translateY(3.5px)',
+        }}
+      />
+    </Flex>
+  );
+};
+
+type FabContentProps = { id: string; paddingLeft: string; paddingRight: string };
+
+const FabContent: React.FC<FabContentProps> = ({ id, paddingLeft, paddingRight }) => {
+  const { signOut } = useCoreClerk();
+
+  return (
+    <Col
+      id='impersonationText'
+      sx={{
+        whiteSpace: 'nowrap',
+        maxWidth: '0px',
+        transition: `max-width 800ms`,
+      }}
+    >
+      <Col
+        sx={{
+          paddingLeft,
+          paddingRight,
+        }}
+      >
+        <Text
+          colorScheme='neutral'
+          variant='regularMedium'
+        >
+          Signed in as {id}
+        </Text>
+        <Link
+          variant='regularMedium'
+          sx={t => ({
+            alignSelf: 'flex-start',
+            color: t.colors.$primary500,
+            ':hover': {
+              cursor: 'pointer',
+            },
+          })}
+          onClick={async () => {
+            await signOut();
+            close();
```

Comment:
❓ I'm actually a [little surprised that this is working](https://developer.mozilla.org/en-US/docs/Web/API/Window/close). Can you please provide a little more context?

Reply:
Think it works just like document (which is the same as window.document). Will change to be explicit though.

Reply:
Hey George, sorry I wasn't clear enough. As far as I know, calling `close()` that way would result in the following console warning and at the same time, that tab *will* remain open
```
Scripts may close only the windows that were opened by them.
```

You can find more details here: https://developer.mozilla.org/en-US/docs/Web/API/Window/close

Reply:
You're right. This happens when the tab was not opened by a script. We can add a redirect in that case.

Reply:
The tab can only be closed if it was opened with `window.open()` by the caller tab, so I'm pretty sure this is not going to work for our use case.

We could just sign the user out  - the host app should already have a 
```
<SignedOut><RedirectToSignIn></SignedOut>
``` 
(or similar) mechanism in place handling signed-out users.

Wdyt @SokratisVidros ?

Reply:
@desiprisg Where did we land on this?

Reply:
We just sign the user out.